src/caches/extrapolation_caches.jl

```diff
@@ -0,0 +1,31 @@
+@cache struct RichardsonEulerCache{uType,rateType,tabType} <: OrdinaryDiffEqMutableCache
+  u::uType
+  uprev::uType
+  tmp::uType
+  k::rateType
+  fsalfirst::rateType
+  tab::tabType
+  T::Array{uType,2}
+end
+
+struct RichardsonEulerConstantCache{Int64} <: OrdinaryDiffEqConstantCache
+  m::Int64 #Generalise type of m
```

Comment:
Is it fine to use Int64 here or should I generalise it?

Reply:
Use `Int` here. `Int` matches the machine integer so it stops a lot of errors that an come up with 32-bit machines